omod/src/main/java/org/openmrs/module/providermanagement/rest/search/ProviderSearchHandler.java

```diff
@@ -0,0 +1,70 @@
+package org.openmrs.module.providermanagement.rest.search;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.providermanagement.Provider;
+import org.openmrs.module.providermanagement.ProviderRole;
+import org.openmrs.module.providermanagement.api.ProviderManagementService;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchConfig;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchHandler;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchQuery;
+import org.openmrs.module.webservices.rest.web.resource.impl.EmptySearchResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Component
+public class ProviderSearchHandler implements SearchHandler {
+
+    protected Log log = LogFactory.getLog(getClass());
+
+    protected final String PROVIDER_ROLES_PARAM = "providerRoles";
+
+    private final SearchConfig searchConfig = new SearchConfig("providerByRole", RestConstants.VERSION_1 + "/provider",
+            Arrays.asList("1.9.* - 9.*"),
+            new SearchQuery.Builder(
+                    "Allows you to find providers by provider role uuid").withRequiredParameters(PROVIDER_ROLES_PARAM).build());
+
+    /**
+     * @see org.openmrs.module.webservices.rest.web.resource.api.SearchHandler#getSearchConfig()
+     */
+    @Override
+    public SearchConfig getSearchConfig() {
+        return searchConfig;
+    }
+
+    /**
+     * @see org.openmrs.module.webservices.rest.web.resource.api.SearchHandler#search(org.openmrs.module.webservices.rest.web.RequestContext)
+     */
+    @Override
+    public PageableResult search(RequestContext context) throws ResponseException {
+  
+        String providerRoleUuids = context.getParameter(PROVIDER_ROLES_PARAM);
+        List<ProviderRole> providerRoles = new ArrayList<ProviderRole>();
+
+        for (String providerRoleUuid : providerRoleUuids.split(",")) {
```

Comment:
I would not expect this to come in via one request param that needs to be manually parsed, but multiple, as request params supports String[] values, not just String values.  eg. the incoming request should be formatted like:

`&providerRole=XXX&providerRole=YYY`

not

`&providerRoles=XXX,YYY`

and then you would be able to get all provider roles via request.getParameter(PROVIDER_ROLES_PARAM) or something like that.

It may be that both are supported by the OOTB behavior, not sure.

Reply:
Yes, I was thinking about this, and, for better or worse, I think the "providerRoles" with commas was the general pattern in other search handlers, but, admittedly, I think I only looked at one.  Let me probe deeper and/or see if we can support both, but I'm lean towards supporting the same pattern we generally use within REST.

Reply:
And i also somehow feel that one request param with comma separated values is not only easier to use but also results into a smaller url. 😊

Reply:
Yeah, looking throughout the REST Web Services code base, we generally use the comma-separated pattern, and the util method provided by the REST module does not support the multi-parameter.

That being said, it was straightforward to update the search handler to support both forms, see updated PR.  For consistency/convention, however, I stuck with the plural form ("providerRoles") of the parameter.